### PR TITLE
Feature ceilometer configuration

### PR DESCRIPTION
**Pull request by guessi**

### CODE REVIEW

File: examples/common.yaml [hunk 1]
```diff
@@ -87,6 +87,8 @@ openstack::neutron::password: 'whi-rtuz'
 openstack::neutron::shared_secret: 'by-sa-bo'
 
 ######## Ceilometer
+openstack::ceilometer::address::management: '192.168.11.4'
```
Comment (cmurphy) [inline examples/common.yaml]: For the example this should be the same as the controller management address, '172.16.33.4'.

File: examples/common.yaml [hunk 2]
```diff
@@ -87,6 +87,8 @@ openstack::neutron::password: 'whi-rtuz'
...
+openstack::ceilometer::mongo::username: 'whi-truz'
```
Comment (cmurphy) [inline examples/common.yaml]: Could this be changed to 'mongo' or something that doesn't match the password?

Comment (cmurphy) [inline examples/common.yaml]: Could you copy these parameters to examples/allinone.yaml as well?

File: manifests/common/ceilometer.pp [hunk 3]
```diff
@@ -3,10 +3,12 @@
 class openstack::common::ceilometer {
   $is_controller = $::openstack::profile::base::is_controller
 
+  $ceilometer_management_address = $::openstack::config::ceilometer_address_management
   $controller_management_address = $::openstack::config::controller_address_management
 
+  $mongo_username = $::openstack::config::ceilometer_mongo_username
   $mongo_password = $::openstack::config::ceilometer_mongo_password
-  $mongo_connection = "mongodb://${controller_management_address}:27017/ceilometer"
+  $mongo_connection = "mongodb://${mongo_username}:${mongo_password}@${ceilometer_management_address}:27017/ceilometer"
```
Comment (cmurphy) [inline manifests/common/ceilometer.pp]: What happens if the user hasn't defined $mongo_username and $mongo_password? Would it make sense to wrap this in `if $mongo_username and $mongo_password` and allow the old mongo connection if they haven't defined the user?

File: manifests/common/ceilometer.pp [hunk 4]
```diff
@@ -3,10 +3,21 @@
...
 
+  $ceilometer_management_address = $::openstack::config::ceilometer_address_management
   $controller_management_address = $::openstack::config::controller_address_management
 
+  $mongo_username = $::openstack::config::ceilometer_mongo_username
   $mongo_password = $::openstack::config::ceilometer_mongo_password
-  $mongo_connection = "mongodb://${controller_management_address}:27017/ceilometer"
+
+  if ! $mongo_username {
+    fail('mongo_username must be set when ceilometer is being configured')
```
Comment (cmurphy) [inline manifests/common/ceilometer.pp]: I would prefer not to hard fail, as this seemed to be working correctly before we were setting the username and password. I think it should use the new connection string if mongo_username and mongo_password are set, and use the old one if they are undef.

Comment (guessi) [inline manifests/common/ceilometer.pp]: thanks for your advices, code committed, please check it again.

**Pull request merged**